manifests/base/core/resources/deployment.yaml

```diff
@@ -73,3 +75,5 @@ spec:
       volumes:
         - name: bundle-cache
```

Comment:
I'm missing the benefit of having `bundle-cache` and `unpack-cache` here, that contain the same bundle contents. Initially the use of `localdir` store was to copy the unpacked contents from pod log (for image source at least) and store locally. 

Disregarding other sources (which we are not looking to support) - the registry client can directly unpack contents at a single location that we look at for generating charts.

Reply:
I totally agree, but my intention behind this PR is to get the functionality in place and we can optimize in the future. catalogd is going to need the same updates in the future, so ideally we can spend some time updating this implementation to be something that can be used in both catalogd and operator-controller in the future.

I'm not against making the changes in this PR, but the scope of this PR is already quite large with the necessary e2e changes and bringing it to parity with catalogd is a good starting point IMO. We can always iterate and improve in the future.